consensus/istanbul/backend/engine_test.go

```diff
@@ -1327,3 +1393,181 @@ func TestGovernance_Votes(t *testing.T) {
+			expected: vote{"kip71.lowerboundbasefee", uint64(20000000000)},
+		},
+		{
+			voting:   vote{"kip71.upperboundbasefee", uint64(500000000000)}, // voted on block 1
+			expected: vote{"kip71.upperboundbasefee", uint64(500000000000)},
+		},
+		{
+			voting:   vote{"kip71.blockgaslimit", uint64(100000000)}, // voted on block 1
+			expected: vote{"kip71.blockgaslimit", uint64(100000000)},
+		},
+		{
+			voting:   vote{"kip71.gastarget", uint64(50000000)}, // voted on block 1
+			expected: vote{"kip71.gastarget", uint64(50000000)},
+		},
+		{
+			voting:   vote{"kip71.basefeedenominator", uint64(32)}, // voted on block 1
+			expected: vote{"kip71.basefeedenominator", uint64(32)},
+		},
+	}
+
+	var configItems []interface{}
+	configItems = append(configItems, proposerPolicy(params.WeightedRandom))
+	configItems = append(configItems, epoch(3))
+	configItems = append(configItems, governanceMode("single"))
+	configItems = append(configItems, blockPeriod(0)) // set block period to 0 to prevent creating future block
+	for _, tc := range testcases {
+		chain, engine := newBlockChain(1, configItems...)
+
+		// test initial governance items
+		assert.Equal(t, uint64(25000000000), chain.Config().Governance.KIP71.LowerBoundBaseFee)
+		assert.Equal(t, uint64(750000000000), chain.Config().Governance.KIP71.UpperBoundBaseFee)
+		assert.Equal(t, uint64(36), chain.Config().Governance.KIP71.BaseFeeDenominator)
+		assert.Equal(t, uint64(84000000), chain.Config().Governance.KIP71.BlockGasLimit)
+		assert.Equal(t, uint64(30000000), chain.Config().Governance.KIP71.GasTarget)
+
+		// add votes and insert voted blocks
+		var (
+			previousBlock, currentBlock *types.Block = nil, chain.Genesis()
+			err                         error
+		)
+
+		engine.governance.SetBlockchain(chain)
+		engine.governance.AddVote(tc.voting.key, tc.voting.value)
+		previousBlock = currentBlock
+		currentBlock = makeBlockWithSeal(chain, engine, previousBlock)
+		_, err = chain.InsertChain(types.Blocks{currentBlock})
+		assert.NoError(t, err)
+
+		// insert blocks until the vote is applied
+		for i := 0; i < 6; i++ {
+			previousBlock = currentBlock
+			currentBlock = makeBlockWithSeal(chain, engine, previousBlock)
+			_, err = chain.InsertChain(types.Blocks{currentBlock})
+			assert.NoError(t, err)
+		}
+
+		govConfig := chain.Config().Governance
+		switch tc.expected.key {
+		case "kip71.lowerboundbasefee":
+			assert.Equal(t, tc.expected.value, govConfig.KIP71.LowerBoundBaseFee)
+		case "kip71.upperboundbasefee":
+			assert.Equal(t, tc.expected.value, govConfig.KIP71.UpperBoundBaseFee)
+		case "kip71.gastarget":
+			assert.Equal(t, tc.expected.value, govConfig.KIP71.GasTarget)
+		case "kip71.blockgaslimit":
+			assert.Equal(t, tc.expected.value, govConfig.KIP71.BlockGasLimit)
+		case "kip71.basefeedenominator":
+			assert.Equal(t, tc.expected.value, govConfig.KIP71.BaseFeeDenominator)
+		default:
+			assert.Error(t, nil)
```

Comment:
It seems redundant

Reply:
You are right but I'd like to see It as a secure coding

Reply:
Why do you think that's secure handling? The test code has definitely five topics and the switch expression handles all that cases.

Reply:
@hyunsooda What I mean is that It is my habit. I think It's safe for `switch` to have `default:` case.
I agree that the code is redundant and doesn't have any security issue in this case.
If someone need more testcase and put invalid key in it, the code would cover the case.

Reply:
This is a unit test. When it comes to making a new unit test, creating a separate function for the corresponding unit test is reasonable.